### PR TITLE
Update release/1.9.x docs to match stable-website

### PR DESCRIPTION
**Pull request by kalafut**
There are [many commits missing](https://github.com/hashicorp/vault/compare/release/1.9.x...stable-website#diff-0eb684c3ea7be1b3de6b0b158c7059da05dff87877c4c40d32ea3a8eaef7a75e) from the `release/1.9.x` branch. This has caused docs merge friction when using the new backport workflow, and this PR is a one time sync from `stable-website` -> `release/1.9.x` to true up the docs content.

### CODE REVIEW

**Comment by kalafut** [comment]
Just realized I need to add a few more items to this (at least the `data` directory)

**Comment by mladlow** [comment]
@alexcarpenter Could you take a look at this PR to make sure everything from https://github.com/hashicorp/vault/pull/13159 looks right on this branch please?

File: website/next.config.js [hunk 1]
```diff
@@ -1,8 +1,20 @@
 const withHashicorp = require('@hashicorp/platform-nextjs-plugin')
 const redirects = require('./redirects.next')
 
+// log out our primary environment variables for clarity in build logs
```
Comment (mladlow) [inline website/next.config.js]: This is in stable website now but not the original PR https://github.com/hashicorp/vault/pull/13159 so not sure whether we want it or not, for example.

**Comment by alexcarpenter** [comment]
@mladlow yes, everything looks right when checking out the website locally on this branch.

**Comment by kalafut** [comment]
@mladlow Are you good to approve?

**Comment by mladlow** [comment]
Yeah, I didn't look at everything with maximum eagle eyes but also don't want to block this.

**Review by mladlow — APPROVED** [review]

**Pull request merged**